packages/devtools_app/lib/src/screens/debugger/breakpoints.dart

```diff
@@ -46,24 +34,36 @@ class _BreakpointsState extends State<Breakpoints>
           itemCount: breakpoints.length,
           itemExtent: defaultListItemHeight,
           itemBuilder: (_, index) {
-            return buildBreakpoint(breakpoints[index], selectedBreakpoint);
+            return _Breakpoint(
+              breakpoint: breakpoints[index],
+              selectedBreakpoint: selectedBreakpoint,
+            );
           },
         );
       },
     );
   }
+}
 
-  Widget buildBreakpoint(
-    BreakpointAndSourcePosition bp,
-    BreakpointAndSourcePosition? selectedBreakpoint,
-  ) {
+class _Breakpoint extends StatelessWidget {
+  const _Breakpoint({
+    required this.breakpoint,
+    required this.selectedBreakpoint,
+  });
+
+  final BreakpointAndSourcePosition breakpoint;
+  final BreakpointAndSourcePosition? selectedBreakpoint;
+
+  @override
+  Widget build(BuildContext context) {
     final theme = Theme.of(context);
-    final isSelected = bp.id == selectedBreakpoint?.id;
+    final isSelected = breakpoint.id == selectedBreakpoint?.id;
+    final controller = screenControllers.lookup<DebuggerController>();
 
     return Material(
       color: isSelected ? theme.colorScheme.selectedRowBackgroundColor : null,
       child: InkWell(
-        onTap: () async => await _onBreakpointSelected(bp),
+        onTap: () async => await _onBreakpointSelected(breakpoint, controller),
```

Comment:
[optional] consider just passing the selectBreakpoint callback instead of the entire contoller